admin/horace_on.m.template

```diff
@@ -10,8 +10,26 @@ function path=horace4_on(non_default_hor_path,non_default_her_path)
 %                                         of Horace/Herbert pairs)
 %
 %
+warn_state=warning('off','all');    % turn of warnings (so don't get errors if remove non-existent paths)
+% Temporary inefficient code, needed only during transition from Horace v3 to Horace V4 on iDAAAS until its config not changed
+try
+    herbert_off();
+catch
+end
+% End temporary code
+try
+    horace_off();
+    warning(warn_state);    % return warnings to initial state
+    evalin('base','clearvars')
+    clearvars -global;
+    clear classes;
+catch
+    warning(warn_state);    % return warnings to initial state
+end
+warning(warn_state);    % return warnings to initial state
```

Comment:
```suggestion
try
    horace_off();
catch
end

warning(warn_state);    % return warnings to initial state

evalin('base','clearvars')
clearvars -global;
clear classes;
```
The reset in the `catch` is unnecessary as it will move onto the next line anyway.

Keep the scope of `try`s as small as possible and only capture potential failures.

Reply:
clearvars will remove `warn_state`  variable, so suggested row 25 will fail

Reply:
I'm not sure what you mean. The `clearvars` happens after line 25. Otherwise the original code would fail.

Reply:
clearing all variables when intitializing Horace is a nuclear options, and we do not want to do it unless it is absolutely necessary. Unfortunately, it appears to be that it is necessary if Horace version have changed. Otherwise, we want to keep our workspaces intact.